### ISSUE TITLE
Python unable to select bands in ReprojectImage or Warp

### ISSUE THREAD

**Issue by constantinius** [OSGeo/gdal#2431]
## Expected behavior and actual behavior.

Currently there is no way (at least not documented) to specify what bands to reproject with the `ReprojectImage` or `Warp` function. There is the possibility to pass additional KVP `options`, but I think they refer to [these options](https://gdal.org/api/gdalwarp_cpp.html#_CPPv4N15GDALWarpOptions16papszWarpOptionsE) and are not band related.

The `GDALWarpOption` struct [provides such a field](https://gdal.org/api/gdalwarp_cpp.html#_CPPv4N15GDALWarpOptions11panSrcBandsE)

## Steps to reproduce the problem.

not applicable

## Operating system

not applicable

## GDAL version and provenance

current master / documentation

**Comment by jratike80** [comment]
I guess that creating an interim VRT that contains the bands that you want in the order that you want and warping the VRT could be a flexible workaround.

This does not feel quite as a bug. Did you consider to send mail to gdal-dev mailing list instead?

**Comment by rouault** [comment]
Yes, neither gdalwarp, and thus nor the C API above it, can set bands to reproject (despite indeed underlying code potentially able to proceed on a subset). Using an intermediate VRT should do the job.

**Comment by constantinius** [comment]
> This does not feel quite as a bug. Did you consider to send mail to gdal-dev mailing list instead?

No, I agree, this is not a bug. I thought using an issue to raise a perceived shortcoming of the API/docs would be the appropriate channel.

> Yes, neither gdalwarp, and thus nor the C API above it, can set bands to reproject (despite indeed underlying code potentially able to proceed on a subset).

I'm confused, doesn't the C API use the `GDALWarpOptions` struct?

As far as I can tell [`rasterio` is using that field](https://github.com/mapbox/rasterio/blob/master/rasterio/_warp.pyx#L524) and it seems to work.

Wouldn't it make sense to also allow it in the vanilla GDAL Python API? I can try to implement this in a PR, but not if you think it does not fit.

**Comment by rouault** [comment]
> I'm confused, doesn't the C API use the `GDALWarpOptions` struct?

Ah, I add in mind the high level GDALWarp() that gdalwarp uses. It has no band selection, but indeed underneath it uses GDALWarpOptions, so that could potentially be added (with some care regarding alpha bands, or when updating an existing target file). And GDALReprojectImage() also uses GDALWarpOptions underneath.

I believe it would be easier (API wise) to add that to GDALWarp() / gdal.Warp() than to GDALReprojectImage(), since GDALWarp() API is extendable (to being mostly command line arguments). A PR would be welcome.
